vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditSlideController.java

```diff
@@ -36,4 +54,47 @@
         slideManager.updateSlide((Slide) slide);
         return new ResponseEntity<String>(HttpStatus.OK);
     }
+
+    @RequestMapping(value="/staff/module/{moduleId}/slide/{slideId}/edit", method=RequestMethod.GET)
+    public String show(Model model, @PathVariable("moduleId") String moduleId, @PathVariable("slideId") String slideId) {
+        ISlide slide = slideManager.getSlide(slideId);
+        SlideForm slideForm = new SlideForm();
+        slideForm.setName(slide.getName());
+        slideForm.setDescription(slide.getDescription());
+        if(slide instanceof BranchingPoint) {
+            slideForm.setType(SlideType.BRANCHING_POINT.toString());
+            IBranchingPoint branchingPoint = (IBranchingPoint) slide;
+            List<String> choices = new ArrayList<String>();
+            for(IChoice choice : branchingPoint.getChoices()) {
+                choices.add(choice.getSequence().getId());
+            }
+            slideForm.setChoices(choices);
+            model.addAttribute("choices", choices);
+        }
+        else {
+            slideForm.setType(SlideType.SLIDE.toString());
+        }
+        model.addAttribute("slideForm", slideForm);
+        model.addAttribute("slideId", slideId);
+        model.addAttribute("moduleId", moduleId);
+        model.addAttribute("sequences", moduleManager.getModuleSequences(moduleId));
+        return "staff/module/slide/edit";
+    }
+
+    @RequestMapping(value="/staff/module/{moduleId}/slide/{slideId}/edit", method=RequestMethod.POST)
+    public String save(@ModelAttribute SlideForm slideForm, @PathVariable("moduleId") String moduleId, @PathVariable("slideId") String slideId) {
+        ISlide slide = slideManager.getSlide(slideId);
+        slide.setName(slideForm.getName());
+        slide.setDescription(slideForm.getDescription());
+        SlideType type = slideForm.getType().isEmpty() ? null : SlideType.valueOf(slideForm.getType());
+        if(type.equals(SlideType.BRANCHING_POINT)) {
+            List<IChoice> choices = choiceFactory.createChoices(slideForm.getChoices());
+            IBranchingPoint branchingPoint = (IBranchingPoint) slide;
+            branchingPoint.setChoices(choices);
```

Comment:
what about the previously created choices?

Reply:
we are providing an option on UI for a user to select and deselect the sequences as choices while editing the slide. When the user changes the list and clicks on save, deselected items will be deleted from Branching choice options and selected ones will be added to the list.

Reply:
where are the previous ones being deleted? you create new choices, set them in the slide, but what happens with the existing choices? Since you don't have orphanremoval set, I don't think hibernate does that for you?

Reply:
got it, It is just deleting and adding the links in the join table by creating new choice ids leaving the existing ones hanging in the Choices List.  I will fix this.